pkg/plugin/framework/server.go

```diff
@@ -14,7 +14,9 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package plugin
+// Package framework is the common package that any plugin client
```

Comment:
Could we move the package-level godoc into its own file (`doc.go`) and maybe include a small code sample of how to use the package (specifically calling the NewServer method, then registering a plugin or two, then serving)?